modules/transcription-service-ibm-watson-impl/src/main/java/org/opencastproject/transcription/ibmwatson/IBMWatsonTranscriptionService.java

```diff
@@ -875,12 +931,85 @@ public void run() {
                     mpId, jobId, e.getClass().getName(), e.getMessage());
           }
         }
+
+        if (maxAttempts > 1) {
+          // Find jobs that need to be re-submitted
+          jobs = database.findByStatus(TranscriptionJobControl.Status.Retry.name());
+          HashMap<String, String> params = new HashMap<String, String>();
+          for (TranscriptionJobControl j : jobs) {
+            String mpId = j.getMediaPackageId();
+            String wfId = startWorkflow(mpId, retryWfDefId, params);
+            String jobId = j.getTranscriptionJobId();
+            if (wfId == null) {
+              logger.warn(
+                      "Retry transcription workflow could NOT be scheduled for mp {}, watson job {}. Will try again next time.",
+                      mpId, jobId);
+              // Will try again next time
+              continue;
+            }
+            logger.info("Retry transcription workflow {} scheduled for mp {}.", wfId, mpId);
+            // Retry was submitted, update previously failed job state to error
+            database.updateJobControl(jobId, TranscriptionJobControl.Status.Error.name());
+          }
+        }
       } catch (TranscriptionDatabaseException e) {
-        logger.warn("Could not read transcription job control database: {}", e.getMessage());
+        logger.warn("Could not read/update transcription job control database.", e);
       }
     }
   }
 
+  private String startWorkflow(String mpId, String wfDefId, Map<String, String> params) {
+    DefaultOrganization defaultOrg = new DefaultOrganization();
+    securityService.setOrganization(defaultOrg);
+    securityService.setUser(SecurityUtil.createSystemUser(systemAccount, defaultOrg));
+
+    // Find the episode
+    final AQueryBuilder q = assetManager.createQuery();
+    final AResult r = q.select(q.snapshot()).where(q.mediaPackageId(mpId).and(q.version().isLatest())).run();
+    if (r.getSize() == 0) {
+      // Media package not archived yet.
+      logger.warn("Media package {} has not been archived yet.", mpId);
+      return null;
+    }
+
+    String org = Enrichments.enrich(r).getSnapshots().head2().getOrganizationId();
+    Organization organization = null;
+    try {
+      organization = organizationDirectoryService.getOrganization(org);
+      if (organization == null) {
+        logger.warn("Media package {} has an unknown organization {}.", mpId, org);
+        return null;
+      }
+    } catch (NotFoundException e) {
+      logger.warn("Organization {} not found for media package {}.", org, mpId);
+      return null;
+    }
+    securityService.setOrganization(organization);
+
+    try {
+      WorkflowDefinition wfDef = workflowService.getWorkflowDefinitionById(wfDefId);
+      Workflows workflows = wfUtil != null ? wfUtil : new Workflows(assetManager, workspace, workflowService);
```

Comment:
Not a fan of this, but I understand why wfUtil exists.  A future rewrite of this level ought to replace this with an injected variable in the tests instead IMO.

Reply:
I am not a fan either. Can you send me an example on how to do it?

Reply:
See how the AssetManager test works.  There's a base class, and then the tests extend the base and implement or override some of the methods.  It's wordier, but it avoids things like this.